mrbgems/mruby-time/src/time.c

```diff
@@ -201,12 +200,15 @@ time_alloc(mrb_state *mrb, double sec, double usec, enum mrb_timezone timezone)
   struct mrb_time *tm;
 
   tm = (struct mrb_time *)mrb_malloc(mrb, sizeof(struct mrb_time));
-  tm->sec  = (time_t)sec;
-  if (sizeof(time_t) == 4 && (sec > (double)INT32_MAX || (double)INT32_MIN > sec)) {
-    goto out_of_range;
-  }
-  else if ((sec > 0 && tm->sec < 0) || (sec < 0 && (double)tm->sec > sec)) {
-  out_of_range:
+  tm->sec = (time_t)sec;
+  if (
+    /* Check for overflow of 32 bit time_t */
+    (sizeof(time_t) == 4 && (sec > (double)INT32_MAX || (double)INT32_MIN > sec)) ||
+    /* Check for overflow of 64 bit time_t */
+    (sizeof(time_t) == 8 && (sec > (double)INT64_MAX || (double)INT64_MIN > sec)) ||
+    /* Check for wrapping of tm->sec */
+    ((sec > 0 && tm->sec < 0) || (sec < 0 && (double)tm->sec > sec))
```

Comment:
I'm not sure, but this line might be a no-op, because signed overflow is undefined behavior.